pkg/operator/utils/utils.go

```diff
@@ -292,79 +294,123 @@ func IsValidMode(operatorMode operatorv1.CloudCredentialsMode) bool {
+		log.Info("Cluster has not successfully completed an installation/upgrade yet")
+		// don't set any condition for this case
+		return nil
+	}
+
+	if !semver.IsValid(clusterSemVer) {
+		upgradeableCondition.Status = configv1.ConditionFalse
+		upgradeableCondition.Reason = constants.ErrorDeterminingUpgradeableReason
+		upgradeableCondition.Message = fmt.Sprintf("Unable to decode cluster version: %s", clusterSemVer)
+		return upgradeableCondition
+	}
+
+	clusterSemVer = semver.MajorMinor(clusterSemVer)
+
+	// If the cluster admin has set the annotation, then bypass the Upgradeable calculation below and assume
+	// the user knows what they are doing.
+	operatorConfig := &operatorv1.CloudCredential{}
+	operatorConfigKey := types.NamespacedName{
+		Name: constants.CloudCredOperatorConfig,
+	}
+	if err := kubeClient.Get(context.Background(), operatorConfigKey, operatorConfig); err != nil {
+		log.WithError(err).Error("unexpected error checking for CloudCredential config annotation")
+		upgradeableCondition.Status = configv1.ConditionFalse
+		upgradeableCondition.Reason = constants.ErrorDeterminingUpgradeableReason
+		upgradeableCondition.Message = fmt.Sprintf("Error determining if cluster can be upgraded to %s: %s",
+			constants.UpgradeableNextRelease, err)
+		return upgradeableCondition
+	}
+
+	upgradeableTo := operatorConfig.Annotations[constants.UpgradeableAnnotation]
+
+	if !strings.HasPrefix(upgradeableTo, "v") {
+		upgradeableTo = "v" + upgradeableTo
+	}
+
+	upgradeableTo = semver.MajorMinor(upgradeableTo)
+
+	if semver.Compare(upgradeableTo, clusterSemVer) == 1 {
+		log.Info("Found annotation indicating upradeability, assuming upgradeable")
+		return nil
+	}
+
+	// Check for upgradeability based on mode
+
 	if mode == operatorv1.CloudCredentialsModeManual {
-		// Check for the existence of credentials we know are coming in the future. If any do not exist, then we cosider
-		// ourselves upgradeable=false and inform the user why.
-		missingSecrets := []types.NamespacedName{}
-		for _, nsSecret := range newSecrets {
-
-			secLog := log.WithField("secret", nsSecret).WithField("toRelease", toRelease)
-			secLog.Debug("checking that secrets exist for manual mode cluster upgrade")
-			secret := &corev1.Secret{}
-
-			if err := kubeClient.Get(context.Background(), nsSecret, secret); err != nil {
-				if errors.IsNotFound(err) {
-					secLog.Info("identified missing secret for upgrade")
-					missingSecrets = append(missingSecrets, nsSecret)
-				} else {
-					// If we can't figure out if you're upgradeable, you're not upgradeable:
-					secLog.WithError(err).Error("unexpected error looking up secret, marking upgradeable=false")
-					upgradeableCondition.Status = configv1.ConditionFalse
-					upgradeableCondition.Reason = constants.ErrorDeterminingUpgradeableReason
-					upgradeableCondition.Message = fmt.Sprintf("Error determining if cluster can be upgradead to %s: %s",
-						toRelease, err)
-					return upgradeableCondition
-				}
-			}
```

Comment:
I'm not sure we should drop this framework, it is somewhat redundant with the annotation check, but it's also an extra layer of security. In general I like the idea of the cco being intelligent in guiding you for what's wrong, and I think covering the case where you may have claimed you minted your creds for next release, but we see a secret that is clearly missing, I think that's worth keeping. WDYT?

Reply:
Then again this violates my above goal about not having to remember to maintain a release number for the next version each release. Curious what you and @akhil-rane think.

Reply:
I'm not sure what the right answer is here. I can see an advanced customer who knows what they're doing not needing to worry about creating a Secret for a component that they have disabled. For exmaple, imagine that the new AWS CSI driver/operator was disableable.
It's a tough needle to thread here as Manual mode was originally intended for more advanced use cases, and going forward we might find that Manual mode might not be that advanced with STS stuff leveraging Manual mode.

On a related note, if we can get people used to running `ccoctl` before upgrades, we can use that opportunity to highlight anything that a user might have missed before upgrade. Someone running `ccoctl` would see the tool lay down the new Secret manifest for the AWS CSI component, so this might help someone having missed a pre-upgrade step.
(all this `ccoctl` talk of course means that the tool would grow to handle more than just the STS use case...)

Reply:
I definitely think ccoctl being able to do more than STS should be on the table eventually, it'd be great for manual mode in general.

The disabled or unused operators is a probably good enough reason to drop this bit of code for now.

Reply:
Maybe I've had a change of heart...on the other hand, we could leave this missing Secret stuff in. Setting the annotation will bypass any missing Secret stuff which allows advanced users to move an upgrade forward.

Reply:
Feels like it may not help, if you must set the annotation, and if setting the annotation overrides the missing secret code, the missing secret code couldn't really ever help. (except before you set the annotation maybe but it's use is pretty thin by that point) I was thinking more as an additional layer of checks after setting the annotation.

Reply:
I prefer manual mode to be "user knows what to do". Even though we can provide additional security by maintaining secrets, we do not provide complete security since permissions can always change. I am fine with removing this code.